app/models/CategoryRecord.scala

```diff
@@ -0,0 +1,112 @@
+  measurementUnit: Option[String] = None
+)
+
+object CategoryRecord {
+
+  implicit lazy val format: OFormat[CategoryRecord] = Json.format
+
+  def build(answers: UserAnswers, eori: String, recordId: String): EitherNec[ValidationError, CategoryRecord] =
+    (
+      getCategory(answers, recordId),
+      answers.getOptionalPageValueForOptionalBooleanPage(
+        answers,
+        HasSupplementaryUnitPage(recordId),
+        SupplementaryUnitPage(recordId)
+      ),
+      getMeasurementUnit(answers)
+    ).parMapN((category, supplementaryUnit, measurementUnit) =>
+      CategoryRecord(
+        eori = eori,
+        recordId = recordId,
+        category = category,
+        supplementaryUnit = supplementaryUnit,
+        measurementUnit = measurementUnit
+      )
+    )
+
+  private val CATEGORY_1 = 1
+  private val CATEGORY_2 = 2
+  private val STANDARD   = 3
+
+  def getCategory(answers: UserAnswers, recordId: String): EitherNec[ValidationError, Int] =
+    answers
+      .get(RecordCategorisationsQuery)
+      .map { recordCategorisations =>
+        recordCategorisations.records
+          .get(recordId)
+          .map { categorisationInfo =>
+            Right(chooseCategory(recordId, answers, categorisationInfo))
+          }
+          .getOrElse(Left(NonEmptyChain.one(RecordIdMissing(RecordCategorisationsQuery))))
+      }
+      .getOrElse(Left(NonEmptyChain.one(PageMissing(RecordCategorisationsQuery))))
+
+  def chooseCategory2Or3(
+    recordId: String,
+    answers: UserAnswers,
+    categorisationInfo: CategorisationInfo
+  ): Int = {
+    val category2AssessmentsCount = categorisationInfo.categoryAssessments.count(_.category == 2)
+    if (category2AssessmentsCount > 0) {
+      answers.getPageValue(AssessmentPage(recordId, categorisationInfo.categoryAssessments.length - 1)) match {
+        case Right(AssessmentAnswer.NoExemption) => CATEGORY_2
+        case Right(_)                            => STANDARD
+        case _                                   => CATEGORY_2
+      }
+    } else {
+      STANDARD
+    }
+  }
+
+  def chooseCategory(
+    recordId: String,
+    answers: UserAnswers,
+    categorisationInfo: CategorisationInfo
+  ): Int = {
+    val category1AssessmentsCount = categorisationInfo.categoryAssessments.count(_.category == 1)
+    if (category1AssessmentsCount > 0) {
+      answers.getPageValue(
+        AssessmentPage(recordId, category1AssessmentsCount - 1)
+      ) match {
+        case Right(AssessmentAnswer.NoExemption) => CATEGORY_1
+        case Right(_)                            => chooseCategory2Or3(recordId, answers, categorisationInfo)
+        case _                                   => CATEGORY_1
+      }
+    } else {
+      chooseCategory2Or3(recordId, answers, categorisationInfo)
+    }
+  }
+
+  //TODO: get measurementUnit from answers
+  private def getMeasurementUnit(answers: UserAnswers): EitherNec[ValidationError, Option[String]] = Right(Some("1"))
```

Comment:
can you put a TBD here once yohan gets this bit done in his ticket, it will be more something like kilograms though